### PR TITLE
feat: add sst-metadata tool to query sst metadata

### PR DESCRIPTION
**Pull request by jiacai2050**
## Related Issues
Closes #

When debug compaction related issue, it's helpful to check sst metadata.

## Detailed Changes
- Add new cli tool `sst-metadata`.

## Test Plan 
Manually
```
$ ./target/release/sst-metadata --dir ~/bench/data/store/2/2199023255817/

Location:2/2199023255817/51491.sst, time_range:[2022-09-05 10:00:00, 2022-09-05 12:00:00), max_seq:305309, size:440.000M, metadata:55.219M, kv:51.565M, filter:38.470M, row_num:14480000
Location:2/2199023255817/53873.sst, time_range:[2022-09-05 10:00:00, 2022-09-05 12:00:00), max_seq:319320, size:666.969M, metadata:83.759M, kv:78.198M, filter:58.342M, row_num:21960000
Location:2/2199023255817/51538.sst, time_range:[2022-09-05 10:00:00, 2022-09-05 12:00:00), max_seq:305001, size:1161.511M, metadata:145.723M, kv:136.025M, filter:101.489M, row_num:38200000
Location:2/2199023255817/53269.sst, time_range:[2022-09-05 10:00:00, 2022-09-05 12:00:00), max_seq:315057, size:1176.751M, metadata:147.631M, kv:137.805M, filter:102.817M, row_num:38700000
Location:2/2199023255817/53973.sst, time_range:[2022-09-05 10:00:00, 2022-09-05 12:00:00), max_seq:320198, size:1183.643M, metadata:148.508M, kv:138.623M, filter:103.428M, row_num:38930000


```

### CODE REVIEW

File: tools/src/bin/sst-metadata.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    let storage = LocalFileSystem::new_with_prefix(args.store_path)?;
+    let storage: ObjectStoreRef = Arc::new(storage);
+    let prefix_path = Path::parse(args.dir)?;
+    let mut ssts = storage.list(Some(&prefix_path)).await?;
+    while let Some(object_meta) = ssts.next().await {
+        let ObjectMeta { location, size, .. } = object_meta?;
+        let md = parse_metadata(&storage, &location, size).await?;
+        let time_range = md.time_range;
+        let seq = md.max_sequence;
+        println!("Location:{location}, time_range:{time_range:?}, size:{size}, max_seq:{seq}");
```
Comment (ShiKaiWi) [inline tools/src/bin/sst-metadata.rs:59]: Shall we print more information here? Such as, number of the rows and row groups, bloom filter size?

Comment (jiacai2050) [reply tools/src/bin/sst-metadata.rs]: fixed

File: Makefile [hunk 2]
```diff
@@ -124,4 +125,8 @@ else
 dev-setup:
 	echo "Error: Unsupported OS. Exiting..."
 	exit 1
-endif
\ No newline at end of file
+endif
+
+fix:
+	cargo fmt
+	cargo sort --workspace
```
Comment (Rachelint) [inline Makefile:132]: Move this to `fmt:`?

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**